src/utils/useNumberParser.ts

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react';
+
+/**
+ * See: https://observablehq.com/@mbostock/localized-number-parsing
+ */
+
+const useNumberParser = (): any => {
+  const parser = (locale: string): any => {
+    const localeToUse = locale === 'gx' ? 'en' : locale;
+    const parts = new Intl.NumberFormat(localeToUse).formatToParts(12345.6);
+    const numerals = new Intl.NumberFormat(localeToUse, { useGrouping: false }).format(9876543210).split('').reverse();
+    const index = new Map(numerals.map((d, i) => [d, i]));
+    const _group = new RegExp(`[${parts?.find((d) => d?.type === 'group')?.value}]`, 'g');
```

Comment:
Should this "group" be the thousands separator? In Argentina it would be a "." but I'm seeing ","

Reply:
Thanks for looking @constanzauanini .
Just so the scope of this PR is clear.. it is to parse a formatted number.
In JS, +'1,000' returns NaN and Number('1,000') returns NaN.
Because of this, some of our Math related code is not functioning like before. (including maybe the totals that you reported as being incorrect). Our withdraw checks, our validation checks.. all use a combination of `+<number-string>` or `Number(<number-string>)`.
This utility's purpose is to return a valid number from a formatted number string.. not the other way around.
So for '1,000' it will return 1000.

Anywhere we see numbers not being formatted in a display - this PR will not fix that.. we need to keep track and fix it separately maybe.

Reply:
Ok, thanks for the clarification!

Reply:
> I found a bug when opening an already created seedling batch, the total quantity is not correct: <img alt="Screenshot 2023-02-08 at 10 02 58" width="655" src="https://user-images.githubusercontent.com/5919083/217538073-9af8e5f6-5ab4-4ae3-996a-d3fac263c434.png">
> 
> Also, that total quantity is not formatted.
> 
> And when withdrawing from one batch the withdrawQuantity and totalReadyQuantity is not formatted: <img alt="Screenshot 2023-02-08 at 10 06 25" width="646" src="https://user-images.githubusercontent.com/5919083/217538549-69b294a4-7c89-4a16-b6c8-bc9454fbd2c6.png">

hmm.. looks like we will need to go through our entire app and clean up how we handle numbers.. if user enters a formatted number etc.. Here the total is incorrect because you entered '2,000' and it was not added to the total count (NaN). we need to fix that for sure..
this is going to be a big sweeping change (not going to be covered in this PR) but something to bring up with @alexTerraformation maybe

Reply:
I entered 2000 when I created, but when I go to edit it, it is formatted like that now

Reply:
But it's not related to this change I guess. It's related to the BE change

Reply:
Got it... I have a change coming up that fixes the total calculation (on all the inventory withdraw views - single, batch, batch-details.. etc) and it also formats the total

Reply:
i have an update coming for just the inventory create/edit/withdraw flow to re-parse number from user entered value.. this will fix the issue..
for rest of app we can plan that separately